tinygrad/shape/view.py

```diff
@@ -317,7 +317,9 @@ def reshape(self, new_shape: Tuple[sint, ...]) -> Optional[View]:
     if self_all_int and not all_int(new_shape):
       if len(self.shape) != len(new_shape): raise ValueError(f"cannot symbolic reshape non-contiguous {self} -> {new_shape}")
       for si, so in zip(self.shape, new_shape):
-        if not isinstance(so, int): so = sym_infer(so, dict([v.unbind() for v in so.vars()]))
+        if not isinstance(so, (int, float)):
+          so = cast(int, sym_infer(so, dict([v.unbind() for v in so.vars()])))
+          if isinstance(so, float): raise ValueError(f"cannot shape to new shape {new_shape=}, contains Variable that infers to float {so}")
```

Comment:
Raised error when `isinstance(so,float)` because `so` has type `SupportsIndex` and float doesn't.